### PR TITLE
Ticket #631 Add component context/actions box to component show page

### PR DESCRIPTION
**Pull request by gordonleacock**
There are three problems remaining:
1. Sometime the document object passed in to the show_actions_box_default partial has no document.unitid (see screenshot below) so no collection id can be displayed. The ticket assume that is always present. FIXED
2. The request button in the action box goes to an error page. I altered the button styles in view/arclight/requests/_google_form.html.erb because it seemed that the action_box will replace the other request button displays but I didn't make any other changes. FIXED
3. I'm having problems getting the styles for the action box correct. They are close but not right. @ggeisler , please review and suggest changes. GARY SUGGESTS WAITING FOR LATER CLEAN UP

No collection id

Kudos to @jkeck for great help with this PR! :-) And to @labradford for helping me get started!

### CODE REVIEW

**Comment by gordonleacock** [comment]
The conflicts noted above are due to a mistaken merge into master of an earlier version of this PR.

**Comment by mejackreed** [comment]
@ggeisler We can alway know which collection it came from, but in this case we use "Collection ID" to be the `unitid`. And unfortunately we can't always count on a `unitid` being available unless we want to enforce that. But I think its optional in the EAD spec.

File: app/views/catalog/_show_default.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class='row'>
   <div class='col-md-12'>
-    <ul class='nav nav-tabs nav-fill' role='tablist' aria-label='<%= t('arclight.views.show.tablist_nav') %>'>
```
Comment (mejackreed) [inline app/views/catalog/_show_default.html.erb]: Why have we changed this markup? This seems to be different than the other design tickets?

File: app/assets/stylesheets/arclight/modules/layout.scss [hunk 2]
```diff
@@ -10,6 +10,22 @@
...
+  font-size: $font-size-sm;
+  margin: 0 (-$spacer) $spacer;
+  padding: 3px $result-item-body-padding;
+
+  .btn-secondary, .al-collection-id, .toggle-bookmark {
+    margin-top: $spacer * .5;
+  }
+  
+  .toggle-bookmark {
+      margin: $spacer ($spacer * .5) (-$spacer * .5) 0;
```
Comment (mejackreed) [inline app/assets/stylesheets/arclight/modules/layout.scss]: Are there some extra spaces in here?

**Comment by estelendur** [comment]
Sometimes the request button does nothing (presumably there is no google form request link) and in that case it should not be present?

File: app/views/catalog/_show_default.html.erb (hunk 1 above)
Comment (gordonleacock) [reply app/views/catalog/_show_default.html.erb]: Arg, fixed. :-)

File: app/assets/stylesheets/arclight/modules/layout.scss (hunk 2 above)
Comment (gordonleacock) [reply app/assets/stylesheets/arclight/modules/layout.scss]: Stupid sublime3, fixed!

**Comment by gordonleacock** [comment]
Okay, 
1. if there is a document.unitid, I show it
2. if there is a document.unitid and document.containers.present?, I show a vertical bar " | "
3. if there is  document.containers.present?, I show document.containers.join(', ')
and
4. I use requestable = item_requestable?('', { document: document }) to condition the display of
<%= render partial: 'arclight/requests/google_form', locals: { document: document } %>
in the hope that this resolves esty's @estelendur comment.

**Comment by gordonleacock** [comment]
I think I've resolved the conflicts and agree that we should make the mobile styling a separate ticket. @ggeisler said he would come back to the currently styling later but I want to note that now, because because both the collection id and the request button may not always be available, that the current styling I added is probably worse. Sorry Gary! :-|

**Comment by gordonleacock** [comment]
I'm getting errors here that I don't get in my local environment with rspec. @mejackreed?

**Comment by mejackreed** [comment]
Have you tried regenerating the test app?

```
rm -rf .internal_test_app
bundle exec rake engine_cart:generate
```

**Comment by gordonleacock** [comment]
Thanks Jack. Now, I'm just getting one error locally:
Failures:

  1) Arclight::RepositoriesController#show looks up the repository detail page
     Failure/Error: expect(collections.first.unitid).to eq 'MS C 271'

       expected: "MS C 271"
            got: "MS C 544"

       (compared using ==)
     # ./spec/controllers/arclight/repositories_controller_spec.rb:34:in `block (3 levels) in <top (required)>'

Finished in 46.16 seconds (files took 3.29 seconds to load)
443 examples, 1 failure, 9 pending

I think Bill has a fix for this kind of thing using find? Not sure...

**Comment by gordonleacock** [comment]
I changed spec/controllers/arclight/repositories_controller_spec.rb to work, let's see if that breaks travis...

File: app/views/catalog/_show_collection.html.erb [hunk 3]
```diff
@@ -32,23 +32,22 @@
...
-      <div class='tab-pane' id='online-content' role='tabpanel'>
-        <%= render partial: 'collection_online_contents', locals: { document: document }  %>
-      </div>
-      <div class='tab-pane' id='access' role='tabpanel'>
-        <% unless blacklight_config.show.context_access_tab_items.nil? %>
-          <% items = blacklight_config.show.context_access_tab_items.select { |i|  fields_have_content?(@document, i) }  %>
-          <% items.each_with_index do |item, index| %>
-            <%= render partial: 'access_contents', locals: { document: @document, field_accessor: item, card_index: index} %>
-          <% end %>
+     <div class='tab-pane active' id='overview' role='tabpanel'>
```
Comment (mejackreed) [inline app/views/catalog/_show_collection.html.erb]: I mentioned this in slack, but dropping it here for other reviewers. I'm not sure about the indention change here and if its: 
  - needed at all
 - indicates any markup changes

If possible, could we indent this block as it was previously?

**Pull request merged**